pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -893,20 +891,19 @@ IdealState ensureAllPartitionsConsuming(TableConfig tableConfig, PartitionLevelS
 
             // Create a new segment to re-consume from the previous start offset
             LLCSegmentName newLLCSegmentName = getNextLLCSegmentName(latestLLCSegmentName, currentTimeMs);
-            long startOffset = latestSegmentZKMetadata.getStartOffset();
+            StreamPartitionMsgOffset startOffset = offsetFactory.create(latestSegmentZKMetadata.getStartOffset());
```

Comment:
We can drop Msg, StreamPartitionOffset

Reply:
A bit late for this (you had your chance in PR#5360 :), it is already in the segment completion protocol, and also in config. I can probably change the class name without damaging compat (I think). I would let this be.